src/transformation/visitors/access.ts

```diff
@@ -16,32 +17,31 @@ export function transformElementAccessArgument(
 
     const type = context.checker.getTypeAtLocation(node.expression);
     const argumentType = context.checker.getTypeAtLocation(node.argumentExpression);
-    if (isNumberType(context, argumentType) && isArrayType(context, type)) {
+    if (isArrayType(context, type) && isNumberType(context, argumentType)) {
```

Comment:
What is the reason for this change? Just ordering them alphabetically?

Reply:
I have changed it to match general evaluation order (first LHS then RHS), but there is no any runtime difference there